### PR TITLE
tests: kernel: Add some testcases for thread

### PR DESCRIPTION
**Pull request by KangJianX**
Add some error condition of testcases to verify whether the
robustness of API. Such as give a NULL to some API and check
the response if get result that we were expacted.

Signed-off-by: Jian Kang <jianx.kang@intel.com>

### CODE REVIEW

File: tests/kernel/threads/thread_error_case/src/test_thread_error.c [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+	 */
+	switch (choice) {
+	case THREAD_START:
+		ztest_set_fault_valid(true);
+		k_thread_start(NULL);
+		break;
+	case FLOAT_DISABLE:
+		ztest_set_fault_valid(true);
+		k_float_disable(NULL);
+		printk("!!!!!!!!!!!!!!!\n");
```
Comment (enjiamai) [inline tests/kernel/threads/thread_error_case/src/test_thread_error.c:55]: Please remove it.

File: tests/kernel/threads/thread_error_case/src/test_thread_error.c [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+#ifdef CONFIG_USERSPACE
+/* TESTPOINT: Pass a null pointer into the API */
+static void test_timeout_remaining_ticks(void)
+{
+	create_negative_test_thread(TIMEOUT_REMAINING_TICKS);
+}
+#else
+static void test_timeout_remaining_ticks(void)
+{
+	ztest_test_skip();
```
Comment (enjiamai) [inline tests/kernel/threads/thread_error_case/src/test_thread_error.c:164]: I think write this way for dedicated few test cases is ok. But it seems like all your test cases in this files are under userspace. If so, my suggestion is remove all this #if-else and modify the testcase.yaml instead.

**Review by enjiamai — CHANGES_REQUESTED** [review]
please take look at my suggestions, thanks.

**Review by LeiW000 — APPROVED** [review]

File: tests/kernel/threads/thread_error_case/src/test_thread_error.c (hunk 2 above)
Comment (KangJianX) [reply tests/kernel/threads/thread_error_case/src/test_thread_error.c]: Yes, I have added filter and remove the duplicate code.

**Review by enjiamai — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**